packages/Discretization/test/tstTpetraVectorView.cpp

```diff
@@ -0,0 +1,89 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#include "DTK_ConfigDefs.hpp"
+
+#include <Kokkos_View.hpp>
+
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_DefaultComm.hpp>
+
+#include <Tpetra_MultiVector.hpp>
+#include <Tpetra_Map.hpp>
+
+//---------------------------------------------------------------------------//
+// Test building and manipulating a tpetra vector with a kokkos view.
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( TpetraVectorView, vector_view, Node )
+{
+    // Template aliases.
+    using SC = double;
+    using LO = int;
+    using GO = unsigned long int;
+    using NO = Node;
+
+    // Get the communicator.
+    auto comm = Teuchos::DefaultComm<int>::getComm();
+
+    // Set the number of global elements.
+    LO num_local = 100;
+    GO num_global = num_local * comm->getSize();
+
+    // Create a map.
+    auto map =
+        Tpetra::createUniformContigMapWithNode<LO,GO,NO>( num_global, comm );
+
+    // Type aliases.
+    using DualViewType =
+        typename Tpetra::MultiVector<SC,LO,GO,NO>::dual_view_type;
```

Comment:
I am not sure I understand the purpose of this test, are we switching to `dual_view`?

Reply:
`Kokkos::DualView` is the required view type used by Tpetra. It creates a host and device copy. They use it to manage data movement between host and device. Computation happens on the device and they use the original host MPI code for communication.